src/js/profiling/convertHermesProfile.ts

```diff
@@ -0,0 +1,94 @@
+import type { StackId, ThreadCpuFrame,ThreadCpuSample, ThreadCpuStack, ThreadId } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import type * as Hermes from './hermes';
+import { parseHermesStackFrameName } from './hermes';
+import type { RawThreadCpuProfile } from './types';
+
+const UNKNOWN_FRAME_FUNCTION_NAME = '<unknown>';
+const UNKNOWN_STACK_ID = -1;
+const JS_THREAD_NAME = 'JavaScriptThread';
+const JS_THREAD_PRIORITY = 1;
+
+/**
+ * Converts a Hermes profile to a Sentry profile.
+ */
+export function convertToSentryProfile(hermesProfile: Hermes.Profile): RawThreadCpuProfile | null {
+  const jsThreads = new Set<ThreadId>();
+  const hermesStacks = new Set<Hermes.StackFrameId>();
+
+  if (hermesProfile.samples.length === 0) {
+    logger.warn('[Profiling] No samples found in profile.');
+    return null;
+  }
+
+  const start = Number(hermesProfile.samples[0].ts);
+  const samples = hermesProfile.samples.map((hermesSample: Hermes.Sample): ThreadCpuSample => {
+    jsThreads.add(hermesSample.tid);
+    hermesStacks.add(hermesSample.sf);
+
+    const elapsed_since_start_ns = (Number(hermesSample.ts) - start) * 1e3;
+
+    return {
+      stack_id: hermesSample.sf,
+      thread_id: hermesSample.tid,
+      elapsed_since_start_ns: elapsed_since_start_ns.toFixed(0),
+    };
+  });
+
+  const hermesFrameChildToParentMap = new Map<Hermes.StackFrameId, Hermes.StackFrameId>();
+  const framesMap = new Map<string, ThreadCpuFrame>();
+  for (const key in hermesProfile.stackFrames) {
```

Comment:
you can early return here if Map.has(key) and skip parsing duplicate frames

Reply:
`hermesStacks` is a set, it contains leafs of the hermes stack tree (root is main func call).

As the leafs are unique, looping over them and walking up the tree to the roor will generate unique stack only.

Or maybe I've overlooked something?